bolt-sidecar/src/chain_io/manager.rs

```diff
@@ -70,15 +73,38 @@ impl BoltManager {
 
             debug!("fetching proposer statuses for chunk {} of {}", i, chunk_count);
 
-            let returndata = match self.0.getProposerStatuses(hashes_chunk).call().await {
-                Ok(returndata) => returndata,
-                Err(error) => {
-                    let decoded_error = utils::try_parse_contract_error(error)
-                        .wrap_err("Failed to fetch proposer statuses from EL client")?;
-
-                    bail!(generate_bolt_manager_error(decoded_error, commitment_signer_pubkey));
+            // Retry the contract call with exponential backoff
+            let returndata = retry_with_backoff(MAX_RETRIES, || async {
+                match self.0.getProposerStatuses(hashes_chunk.clone()).call().await {
+                    Ok(data) => Ok(data),
+                    Err(err) => {
+                        match &err {
+                            Error::TransportError(RpcError::Transport(transport_err)) => {
+                                if transport_err.is_retry_err() {
+                                    warn!(
+                                        "Retryable transport error when connecting to EL node: {}",
+                                        transport_err
+                                    );
+                                    Err(eyre::eyre!("Transport error: {}", transport_err))
+                                } else {
+                                    bail!("Non-retryable transport error when connecting to EL node: {}", transport_err)
+                                }
+                            }
+                            _ => {
+                                // For other errors, parse and return immediately
+                                let decoded_error = utils::try_parse_contract_error(err)
+                                    .wrap_err("Failed to fetch proposer statuses from EL client")?;
+
+                                bail!(generate_bolt_manager_error(
+                                    decoded_error,
+                                    commitment_signer_pubkey,
+                                ))
+                            }
+                        }
+                    }
```

Comment:
Here you should be able to avoid a level of indentation by matching the inner enum directly, like so:
```rs
Err(Error::TransportError(RpcError::Transport(err)) => { /* ... */ }
Err(_) => { /* ... */ }
```